File: Java/src/com/company/TwoPointer/LC167_TwoSumII.java
package com.company.TwoPointer;

import java.util.ArrayList;

/*
    lc 167
    two ptrs

 */
public class LC167_TwoSumII {
    public int[] twoSum(int[] numbers, int target) {
        int[] res = new int[2];
        int l = 0, r = numbers.length-1;
        while(r > l){
            long a = numbers[l], b = numbers[r];
            if(a + b > target) r--;
            else if(a + b < target) l++;
            else break;
        }

        res[0] = l + 1;
        res[1] = r + 1;

        return res;
    }

    public static void main(String[] args) {
        int[] numbers = {2,5,7,15};
        int target = 9 ;
        LC167_TwoSumII T = new LC167_TwoSumII();
        int[] res = T.twoSum(numbers, target);
        System.out.println(res[0] + " " +  res[1]);
    }

}

File: python/Array/LC167_twoSumII.py
class Solution:
    #      def twoSum(self, numbers: List[int], target: int) -> List[int]:
    #         # binary search O(nlogn) time O(1) space
    #         for i in range(len(numbers)-1):
    #             rest = target - numbers[i]
    #             match_idx = self.search(numbers, i+1, rest)
    #             if match_idx != -1:
    #                 res=[i+1, match_idx+1]
    #                 return res
    #         return res

    #     def search(self, numbers,start, target):
    #         l = start
    #         h = len(numbers)-1
    #         while(l<=h):
    #             m = l+(h-l)//2
    #             if(numbers[m] > target):
    #                 h = m-1
    #             elif(numbers[m] < target):
    #                 l = m+1

    #             else:
    #                 return m
    #         return -1
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        # two pointer O(n) time O(1) space
        l = 0
        h = len(nums)-1
        while(l <= h):
            sums = nums[l]+nums[h]
            if(sums > target):
                h -= 1
            elif(sums < target):
                l += 1
            else:
                return [l+1, h+1]
        return -1
